Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -928,9 +930,11 @@ extension RouteController: ReroutingControllerDelegate {
         }
     }
     
-    func rerouteControllerDidRecieveReroute(_ rerouteController: RerouteController, response: RouteResponse, options: RouteOptions) {
-        updateRoute(with: IndexedRouteResponse(routeResponse: response,
-                                               routeIndex: 0),
+    func rerouteControllerDidRecieveReroute(_ rerouteController: RerouteController, response: RouteResponse, options: RouteOptions, routeOrigin: RouterOrigin) {
+        var indexedRouteResponse = IndexedRouteResponse(routeResponse: response,
```

Comment:
Should we add an internal constructor with the `responseOrigin` property?

Reply:
Good idea, thanks